lib/core/repository/categories/cms.rb

```diff
@@ -0,0 +1,29 @@
+module Core::Repository
+  module Categories
+    class CMS < Core::Repository::Base
+      def initialize(options = {})
+        self.connection = Core::Registry::Connection[:cms]
+        self.fallback_repository = options[:fallback]
+      end
+
+      def all
+        response = connection.get('/%{locale}/categories.json' % { locale: I18n.locale })
```

Comment:
URI template (RFC 6570) would be nice anywhere you perform these kind of expansions. There are loads of different gems that do template expansion. I've used `uri_template` before.